src/native/managed/cdacreader/src/Contracts/RuntimeTypeSystem_1.cs

```diff
@@ -698,14 +698,15 @@ public bool IsArrayMethod(MethodDescHandle methodDescHandle, out ArrayFunctionTy
             return false;
         }
 
-        int arrayMethodIndex = methodDesc.Slot - GetNumVtableSlots(GetTypeHandle(methodDesc.MethodTable));
+        MethodTable methodTable = _methodTables[GetTypeHandle(methodDesc.MethodTable).Address];
```

Comment:
This is kind of subtle.  It's some kind of a `private MethodTable EnsureCreated(TargetPointer methodTableAddress)` - you can almost write `_methodTables[methodDesc.MethodTable]` but we wouldn't have gone through the validation and inserting the MethodTable into the cache.